pkg/test/httprecorder/request_log.go

```diff
@@ -53,12 +57,34 @@ func (r *Request) FormatHTTP() string {
 	}
 	b.WriteString("\n")
 	if r.Body != "" {
-		b.WriteString(r.Body)
+		body := resetTimestamp(r.Body)
+		b.WriteString(body)
 		b.WriteString("\n\n")
 	}
 	return b.String()
 }
+func resetTimestamp(body string) string {
```

Comment:
This is a nice way to do things, but let's not do it here.  Instead can we move it a method like `ReplaceURLPrefix`?  The idea is that we capture the accurate log, and then deliberately apply a few normalizations, doing it that way should make debugging a bit easier because we can still see the "real" values.

The other way to do with time is to have a "Clock" interface, and have the default implementation return the time using `time.Now`, but have a version we use in tests that returns a controlled value.

The nice thing about your normalization approach is that we can more easily test against a real kubeapiserver also, so I like your approach better than injecting a mock Clock :-)

Let's just move the logic to a "fixer" function

Reply:
you are right. I didn't think carefully about the test maintenance cost. Totally agree it could be hard to debug if the request/response body is modified and the logic is buried somewhere not directly reflected in the `_test.go` code.
Updated